ZFSin/zfs/lib/libefi/rdwr_efi.c

```diff
@@ -139,7 +139,7 @@ efi_crc32(const unsigned char *buf, unsigned int size)
 }
 
 static int
-read_disk_info(int fd, diskaddr_t *capacity, uint_t *lbsize)
+read_disk_info(uint64_t fd, diskaddr_t *capacity, uint_t *lbsize)
```

Comment:
This one is tricky. libefi is a library we got from ZOL, and due to that, it uses posix `"int fd"` for file-descriptors. I wrote the hacky "wosix" wrapper around that `HANDLE`s will "often" fit inside an `int`, and we convert (truncate) between the two using `HTOI()` and `ITOH()` - which should "hide" any warnings wrt to truncation.
We could claim ownership of libefi, and make it "ours", but much of libzfs also uses `int fd`, and changing that could make future merges painful. I can think of a few way to address this though.

Which leads to a bigger discussion on how we should handle that in future.